metagpt/provider/azure_openai_api.py

```diff
@@ -43,8 +45,36 @@ def _make_client_kwargs(self) -> dict:
         return kwargs
 
     def _calc_usage(self, messages: list[dict], rsp: str) -> CompletionUsage:
```

Comment:
Noting that there are already 4 places in similar logic, I suggest providing a common function that abstracts the common parts

Reply:
`_calc_usage` is related to the service quotes provided by the service providers and remains the responsibility of the LLM provider to maintain independently; 
the logic for `_update_costs` is consolidated under `BaseLLM` for unified maintenance.